src/transaction_signature_options/sign_with_keychain/mod.rs

```diff
@@ -218,6 +258,47 @@ impl interactive_clap::FromCli for SignKeychain {
     {
         let mut clap_variant = optional_clap_variant.unwrap_or_default();
 
+        let signer_public_key: crate::types::public_key::PublicKey = if context.offline {
```

Comment:
Make `signer_public_key` as an optional parameter like nonce/block_hash/block_height, and move this code to the `input_signer_public_key` function.

Reply:
Done